activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,20 @@
+*   Deprecate `:class_name` option for the `:through` association
+
+    Through association should always relay on the `:class_name` of its source
+    association. Example:
+
+    ```
+    Post.has_many :commenters, through: :comments,
+      source: :author, class_name: 'CommentAuthor'
+    post.comments.first.author.class # => Author
+    post.commenters.first.class # => CommentAuthor
+    ```
+
+    That causes inconsistency to the through association general sense and
```

Comment:
I'm not sure that I fully understand this statement. Specifically, I'm not sure I understand what is meant by ". . .that assumes that principle to be true." I made a couple of rewording suggestions:
```suggestion
    Using `:class_name` causes inconsistencies to the through association including
    some bugs in preloading.
```

Reply:
The principle is (by my example in the description): `post.commenters == post.flat_map(&:comments).map(&:author)` when the association has no custom scope. Please help me to describe that.